chapters/scoping.tex

```diff
@@ -91,8 +91,10 @@ \subsection{Composite Name Lookup}\label{composite-name-lookup}
 \item
   If the first identifier denotes a component, the rest of the name (e.g., \lstinline!B! or \lstinline!B.C!) is looked up among the declared named component elements of the component.
 \item
-  If not found, and if the first identifier denotes a scalar component, or \lstinline!component[j]! where \lstinline!component! is an array of components and the indices \lstinline!j! can be evaluated at translation time and \lstinline!component[j]! is a scalar; and if the composite name is used as a function call, the lookup is also performed among the declared named class elements of the scalar component, and must find a non-operator function.
+  If not found, and if the first identifier denotes a component and the composite name is used as a function call, the lookup is also performed among the declared elements of the component, and must find a non-operator function.
```

Comment:
Is there a corresponding rule somewhere else that allow the same kind of lookup when using functions in other ways (in particular when passed as argument to functions), or do we have different lookup rules for functions depending on how they are used?

Reply:
@henrikt-ma, I understand the original text wants to make sure that the composite function is unique, because you either find it inside a scalar component, or in a scalar element of an array component. Otherwise, if component arrays are involved,  it won't be clear which one is the function to call. Why are you removing that part?

Reply:
I am not removing anything. I'm just asking a question.

Reply:
Edit: why would you like to remove that part. The question remains 😄

Reply:
For functions passed as arguments to functions this cannot occur, unless we start considering functions passed as part of objects to functions (which would lead towards the path of object-oriented dynamic dispatch). This change has no impact on that.

Reply:
By _cannot occur_, do you mean that while a plain function call is possible,
```
x = c1.c2.c3.A.B.function(1.5);
```
we don't allow
```
x = apply(c1.c2.c3.A.B.function, 1.5);
```
?

If so, that is a simple "yes we have different lookups" answer to my question, but of course the real question is then why the two lookups should be different?

(Not saying we need to reopen the issue, just trying to sort out this conversation.)

Reply:
What I am reacting to is this: _and the composite name is used as a function call_